api/institutions/views.py

```diff
@@ -0,0 +1,188 @@
+from rest_framework import generics
+from rest_framework import permissions as drf_permissions
+
+from modularodm import Q
+
+from framework.auth.oauth_scopes import CoreScopes
+
+from website.models import Institution, Node, User
+
+from api.base import permissions as base_permissions
+from api.base.filters import ODMFilterMixin
+from api.base.views import JSONAPIBaseView
+from api.base.utils import get_object_or_error
+from api.nodes.serializers import NodeSerializer
+from api.users.serializers import UserSerializer
+
+from .serializers import InstitutionSerializer
+
+class InstitutionMixin(object):
+    """Mixin with convenience method get_institution
+    """
+
+    institution_lookup_url_kwarg = 'institution_id'
+
+    def get_institution(self):
+        inst = get_object_or_error(
+            Institution,
+            self.kwargs[self.institution_lookup_url_kwarg],
+            display_name='institution'
+        )
+        return inst
+
+
+class InstitutionList(JSONAPIBaseView, generics.ListAPIView):
```

Comment:
This should be filterable, so see about adding the ODM filter mixin from api/base/filters.py